packages/yoga/src/Button/native/Icon.jsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import styled, { withTheme } from 'styled-components';
+import { node, oneOfType, func, bool } from 'prop-types';
+import Icon from '../../Icon';
+import withTouchable from './withTouchable';
+
+const ButtonContainer = styled.View`
+  ${({
+    large,
+    secondary,
+    theme: {
+      yoga: {
+        components: { button },
+      },
+    },
+    disabled,
+    inverted,
+    pressed,
+  }) => {
+    const state = secondary ? 'secondary' : 'primary';
+
+    return `
+      display:flex;
+      align-items: center;
+      justify-content: center;
+      
+      border-radius: ${large ? 24 : 12}px;
+
+      background-color: ${
+        button.types.contained.backgroundColor[state].default
+      };
+     
+      width: ${
+        large ? button.types.icon.size.large : button.types.icon.size.default
+      }px;
+      height: ${
+        large ? button.types.icon.size.large : button.types.icon.size.default
+      }px;
+
+
+      ${
+        !disabled && pressed
+          ? `
+        background-color: ${button.types.contained.backgroundColor[state].pressed};
+      `
+          : ''
+      }
+
+      ${
+        disabled
+          ? `
+        background-color: ${button.types.contained.backgroundColor.disabled};
+      `
+          : ''
+      }
+
+      ${
+        inverted && !disabled
+          ? `
+          background-color: ${button.types.contained.font.default.color};
+        `
+          : ''
+      }
+    `;
+  }}
+`;
+
+const ButtonIcon = ({
```

Comment:
Should we export ButtonIcon, on Native, using `forwardRef` too?
You guys are doing it just for Web